test/common/filter/tcp_proxy_test.cc

```diff
@@ -618,6 +624,32 @@ TEST_F(TcpProxyTest, AccessLogUpstreamHost) {
   EXPECT_EQ(access_log_data_, "127.0.0.1:80 fake_cluster");
 }
 
+TEST_F(TcpProxyTest, AccessLogUpstreamLocalAddress) {
```

Comment:
One other small nit: We are trying now to add small comments above each test to say what they do. If you could do that, that would be great. A follow up is fine also.

Reply:
Done